wrappers/python/openmm/app/xtcfile.py

```diff
@@ -0,0 +1,156 @@
+from __future__ import print_function, division, absolute_import
+
+__author__ = "Raul P. Pelaez"
+__version__ = "1.0"
+
+from openmm.app.internal.xtc_utils import (
+    read_xtc,
+    read_xtc_frames,
+    xtc_write_frame,
+    get_xtc_nframes,
+    get_xtc_natoms,
+)
+import numpy as np
+import os
+from openmm import Vec3
+from openmm.unit import nanometers, femtoseconds, is_quantity, norm
+import math
+import tempfile
+import shutil
+
+
+class XTCFile(object):
+
+    """Parses an XTC file and provides access to the coordinates, box
+    vectors, time and step of each  frame.
+    This class also provides a method  to write new frames into an XTC
+    file from a set of coordinates, box vectors, time and step.
+    """
+
+    def __init__(self, file, topology, dt, firstStep=0, interval=1, append=False):
+        """Create a XTC file, or open an existing file to append.
+
+        Parameters
+        ----------
+        file : file
+            A file to write to
+        topology : Topology
+            The Topology defining the molecular system being written
+        dt : time
+            The time step used in the trajectory
+        firstStep : int=0
+            The index of the first step in the trajectory
+        interval : int=1
+            The frequency (measured in time steps) at which states are written
+            to the trajectory
+        append : bool=False
+            If True, open an existing XTC file to append to.  If False, create a new file.
+        """
+        self._filename = file.name
```

Comment:
I don't think we can depend on this working correctly.  Depending on how the file object was created, `name` might be a relative path, an absolute path, an integer referencing an OS-level file descriptor, or even just a human readable label like `'<stdout>'`.  If it was created by the `tempfile` module, there might not even exist any path for it.

Since XTCReporter already takes a filename instead of an open file object, and that's how most people will access it, it would probably be fine to just make this take a filename as well.